### PR TITLE
Improve core functions test coverage

### PR DESCRIPTION
**Pull request by kloon**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds unit tests for core function that currently do not have any tests.

### How to test the changes in this Pull Request:

1. Run `vendor/bin/phpunit tests/unit-tests/util/class-wc-tests-core-functions.php`
2. Ensure all the tests pass

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: tests/unit-tests/util/class-wc-tests-core-functions.php [hunk 1]
```diff
@@ -694,4 +725,101 @@ public function test_wc_selected() {
...
+				'attr2' => 'Attr2 Value 1',
+				'attr1' => 'Attr1 Value 2',
+			),
+			array(
+				'attr2' => 'Attr2 Value 2',
+				'attr1' => 'Attr1 Value 2',
+			),
+		);
+
+		$this->assertEquals( $expected_combinations, array_reverse( wc_array_cartesian( $variation_attributes ) ) );
```
Comment (claudiulodro) [inline tests/unit-tests/util/class-wc-tests-core-functions.php]: This test [seems to fail sometimes](https://travis-ci.org/woocommerce/woocommerce/builds/402659882?utm_source=github_status&utm_medium=notification). There may be a bug in `wc_array_cartesian` where the order isn't the same every time or (more likely) `$variation_attributes` needs to be sorted before passing into `wc_array_cartesian`.

**Review by claudiulodro — CHANGES_REQUESTED** [review]
Good idea! This looks great; just a failing test that needs investigation.

File: tests/unit-tests/util/class-wc-tests-core-functions.php (hunk 1 above)
Comment (peterfabian) [reply tests/unit-tests/util/class-wc-tests-core-functions.php]: If this is a unit-test for function `wc_array_cartesian`, then I think we should pass two predefined arrays to the function and check the result and not make it depend on results of another, unrelated function.

But it's nice that these irregularities surface when writing tests..

File: tests/unit-tests/util/class-wc-tests-core-functions.php [hunk 2]
```diff
@@ -694,4 +725,101 @@ public function test_wc_selected() {
...
+	 * @return void
+	 */
+	public function test_wc_get_credit_card_type_label() {
+		$this->assertEquals( 'Visa', wc_get_credit_card_type_label( 'visa' ) );
+		$this->assertEquals( 'JCB', wc_get_credit_card_type_label( 'jCb' ) );
+		$this->assertEquals( 'MasterCard', wc_get_credit_card_type_label( 'Mastercard' ) );
+		$this->assertEquals( 'American Express', wc_get_credit_card_type_label( 'american_express' ) );
+		$this->assertEquals( 'American Express', wc_get_credit_card_type_label( 'american-express' ) );
+	}
+
```
Comment (peterfabian) [inline tests/unit-tests/util/class-wc-tests-core-functions.php]: I think this one should test at least some edge cases, e.g. passing empty or string that does not represent a known CC name to the function.

**Review by claudiulodro — APPROVED** [review]

**Comment by claudiulodro** [comment]
Looks good. 👍 Feel free to merge after resolving merge conflicts.

**Pull request merged**